pkg/aws/client.go

```diff
@@ -473,41 +414,40 @@ func (b *ClientBuilder) Build() (Client, error) {
 	return c, err
 }
 
-func (c *awsClient) GetIAMCredentials() (a.Credentials, error) {
+func (c *awsClient) GetIAMCredentials() (aws.Credentials, error) {
 	cfg, err := config.LoadDefaultConfig(context.TODO())
 	if err != nil {
-		return a.Credentials{}, err
+		return aws.Credentials{}, err
 	}
 
 	creds, err := cfg.Credentials.Retrieve(context.TODO())
 	if err != nil {
-		return a.Credentials{}, err
+		return aws.Credentials{}, err
 	}
 	return creds, nil
 }
 
 func (c *awsClient) GetRegion() string {
-	return aws.StringValue(c.awsSession.Config.Region)
+	return c.cfg.Region
 }
 
-func (c *awsClient) ListSubnets(subnetIds ...string) ([]*ec2.Subnet, error) {
+func (c *awsClient) ListSubnets(subnetIds ...string) ([]ec2types.Subnet, error) {
```

Comment:
Lots of places we are switching from pointer, why is that?

Reply:
It was per Ori's request when he reviewed the initial pr: https://github.com/openshift/rosa/pull/1564

Reply:
cheers thanks